conda_build/inspect_pkg.py

```diff
@@ -87,23 +93,21 @@ def __str__(self):
 untracked_package = _untracked_package()
 
 
+@deprecated.argument("3.28.0", "4.0.0", "package", rename="subdir")
+@deprecated.argument("3.28.0", "4.0.0", "prepend")
+@deprecated.argument("3.28.0", "4.0.0", "minimal_hint")
 def check_install(
-    packages, platform=None, channel_urls=(), prepend=True, minimal_hint=False
-):
-    prefix = tempfile.mkdtemp("conda")
-    try:
-        specs = specs_from_args(packages)
-        index = get_index(
-            channel_urls=channel_urls, prepend=prepend, platform=platform, prefix=prefix
-        )
-        actions = install_actions(
-            prefix, index, specs, pinned=False, minimal_hint=minimal_hint
-        )
-        display_actions(actions, index)
-        return actions
-    finally:
-        rm_rf(prefix)
-    return None
+    packages: Iterable[str],
+    subdir: str | None = None,
+    channel_urls: Iterable[str] = (),
+) -> None:
+    with TemporaryDirectory() as prefix:
+        Solver(
+            prefix,
+            channel_urls,
+            [subdir],
```

Comment:
Will this fail if `subdir` is `None`? I'd make sure the default value is compatible with the `Solver` class.

Reply:
Also, I remember that conda-build would merge `native_subdir, "noarch"` in a single made-up subdir internally, so I don't know if this assumption was here too (in case we need to pass `"noarch"` here). I think it doesn't, but I'd double check just in case.

Reply:
subdir fallback: https://github.com/conda/conda-build/pull/5033/commits/9c1100f918643817f39656a1f6b82c64c83e51a9